lms/product/canvas/_plugin/misc.py

```diff
@@ -37,10 +38,23 @@ def post_launch_assignment_hook(self, request, js_config, assignment):
         if focused_user := request.params.get("focused_user"):
             js_config.set_focused_user(focused_user)
 
+    def get_assignment_configuration(
+        self,
+        request,
+        assignment: Assignment | None,
+        historical_assignment: Assignment | None,
+    ) -> AssignmentConfig:
+        document_url = self._get_document_url(request)
+
+        return {
+            "document_url": document_url,
+            # For canvas we add parameter to the launch URL as we don't store the
+            # assignment during deep linking.
+            "group_set_id": request.params.get("group_set"),
```

Comment:
This replace the implementation of `get_group_set_id` over lms/product/canvas/_plugin/grouping.py